grails-app/conf/application.yml

```diff
@@ -253,37 +263,29 @@ environments:
     development:
         dataSource:
             dumpQueriesOnException: true
-            formatSql: true
-            includeInnodbStatusInDeadlockExceptions: true
             logAbandoned: true
-            logSql: false
             logSlowQueries: true
             username: "openboxes"
             password: "openboxes"
-            url: jdbc:mysql://localhost:3306/openboxes?serverTimezone=UTC&useSSL=false
+            # P6Spy is an interceptor that allows us to log database queries. We use it in our Sentry integration.
+            driverClassName: com.p6spy.engine.spy.P6SpyDriver
+            url: jdbc:p6spy:mysql://localhost:3306/openboxes?serverTimezone=UTC&useSSL=false
     test:
         dataSource:
             dumpQueriesOnException: true
-            formatSql: false
-            includeInnodbStatusInDeadlockExceptions: true
             logAbandoned: true
-            logSql: false
-            logSlowQueries: false
             # Our integration tests run against a containerized database: https://java.testcontainers.org/modules/databases/jdbc/.
             # Note that testcontainers auto-disable SSL when connecting via jdbc: https://github.com/testcontainers/testcontainers-java/pull/561
             # For more information, see /src/integration-test/README.md
             url: jdbc:tc:${TEST_DATABASE:mysql:5.7.44}:///openboxes?TC_MY_CNF=testcontainers
             driverClassName: org.testcontainers.jdbc.ContainerDatabaseDriver
     production:
         dataSource:
-            dumpQueriesOnException: false
-            formatSql: false
-            includeInnodbStatusInDeadlockExceptions: true
-            logSql: false
-            logSlowQueries: false
             username: "openboxes"
             password: "openboxes"
-            url: jdbc:mysql://localhost:3306/openboxes?serverTimezone=UTC&useSSL=false
+            # P6Spy is an interceptor that allows us to log database queries. We use it in our Sentry integration.
+            driverClassName: com.p6spy.engine.spy.P6SpyDriver
+            url: jdbc:p6spy:mysql://localhost:3306/openboxes?serverTimezone=UTC&useSSL=false
```

Comment:
let's discuss this. All this should be doing is adding an interceptor to capture the SQL queries, which then get forwarded on to the database as normal. Configuring it this way (via jdbc URL) is the recommended approach, though it does feel a bit weird.

Obviously we'll test it on a deployed environment but let me know if there's something that I'm missing about our setup that would prevent us from doing this.